### ISSUE TITLE
feature: include non-breaking hyphens in run.text

### ISSUE THREAD

**Issue by stefan-hock20** [python-openxml/python-docx#1018]
non braking hyphons are ignored leaving a sequence like: 1-24 hour as 124 hours.

original text in word document:
This occurred 1-24 hours after administration of sugar mainly at timepoints ≥12 hours.
exracted text:
This occurred 124 hours after administration of sugar mainly at timepoints ≥12 hours.

**Comment by scanny** [comment]
@stefan-hock20 please show the minimum reproducible code and also dump the XML for (I suppose) the paragraph in question with `print(paragraph._p.xml)`.

**Comment by scanny** [comment]
Hmm, okay, so there is a `<w:noBreakHyphen/>` element in the run in question and the `.text` property doesn't recognize that.

I think something like this would get it done, basically a local reimplementation of `Paragraph.text`.

```python
from docx.oxml.ns import qn


def paragraph_text(paragraph):
    """Return str text contained in `paragraph`."""
    return "".join(run_text(run) for run in paragraph.runs)


def run_text(run):
    """Return str text contained in `run`."""

    def iter_text_runs():
        for child in run._r:
            if child.tag == qn("w:t"):
                yield child.text if child.text is not None else ""
            elif child.tag == qn("w:tab"):
                yield "\t"
            elif child.tag in (qn("w:br"), qn("w:cr")):
                yield "\n"
            elif child.tag == qn("w:noBreakHyphen"):
                yield "-"

    return "".join(iter_text_runs())


print(paragraph_text(paragraph))
```

**Comment by scanny** [comment]
Non-breaking hyphens are included in paragraph and run text starting in v1.0.0, circa Oct 12, 2023.
